compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -814,7 +817,8 @@ private Result seekMatch(ParserRuleContext currentCtx, int lookahead, int curren
                 case ACCESS_EXPRESSION:
                     return seekInAccessExpression(currentCtx, lookahead, currentDepth, matchingRulesCount);
                 case BASIC_LITERAL:
-                    hasMatch = isBasicLiteral(nextToken.kind);
+                    STToken nextNextToken = this.tokenReader.peek(lookahead + 1);
+                    hasMatch = isBasicLiteral(nextToken.kind, nextNextToken.kind);
```

Comment:
Don't need to handle `()` in basic literals.. It was added to deal with literals with a single token. So may be we could only handle `null` here.

For `()` lets use `NIL_LITERAL` and add it to the `EXPRESSIONS` alternatives array.

Reply:
Adding `NIL_LITERAL` to `EXPRESSIONS` array cause 
```public function foo hello (
}``` to be generated as ```public function foo  MISSING[(]hello  MISSING[] MISSING[=](
 MISSING[] MISSING[)] MISSING[)] MISSING[{]}```
I will have a look on what's happening there.

Reply:
Adding `NIL_LITERAL` to `EXPRESSIONS` array cause func definition with extra func name;  `public function foo hello (
}` to be generated as

`public function foo  MISSING[(]hello  MISSING[] MISSING[=](
 MISSING[] MISSING[)] MISSING[)] MISSING[{]}`

I will have a look on what's happening there